OpacityCore/src/main/kotlin/com/opacitylabs/opacitycore/InAppBrowserActivity.kt

```diff
@@ -20,41 +20,75 @@ import org.mozilla.geckoview.GeckoView
 import org.mozilla.geckoview.WebExtension
 
 class InAppBrowserActivity : AppCompatActivity() {
-    private val closeReceiver = object : BroadcastReceiver() {
+    private val closeReceiver =
+        object : BroadcastReceiver() {
+            override fun onReceive(context: Context?, intent: Intent?) {
+                if (intent?.action == "com.opacitylabs.opacitycore.CLOSE_BROWSER") {
+                    finish()
+                }
+            }
+        }
+
+    private val cookiesForCurrentURLRequestReceiver =
+        object : BroadcastReceiver() {
+            override fun onReceive(context: Context?, intent: Intent?) {
+                if (intent?.action == "com.opacitylabs.opacitycore.GET_COOKIES_FOR_CURRENT_URL"
+                ) {
+                    val receiver = intent.getParcelableExtra<CookieResultReceiver>("receiver")
+                    val domain = java.net.URL(currentUrl).host
+                    val browserCookies = cookies[domain] ?: JSONObject()
+                    receiver?.onReceiveResult(browserCookies)
+                }
+            }
+        }
+
+    private val cookiesForDomainRequestReceiver = object : BroadcastReceiver() {
         override fun onReceive(context: Context?, intent: Intent?) {
-            if (intent?.action == "com.opacitylabs.opacitycore.CLOSE_BROWSER") {
-                finish()
+            if (intent?.action == "com.opacitylabs.opacitycore.GET_COOKIES_FOR_DOMAIN") {
+                val receiver = intent.getParcelableExtra<CookieResultReceiver>("receiver")
+                val domain = intent.getStringExtra("domain")
+                val browserCookies = cookies[domain] ?: JSONObject()
+                receiver?.onReceiveResult(browserCookies)
             }
         }
     }
 
     private lateinit var geckoSession: GeckoSession
     private lateinit var geckoView: GeckoView
-    private var browserCookies: JSONObject = JSONObject()
+    private var cookies: MutableMap<String, JSONObject> = mutableMapOf()
     private var htmlBody: String = ""
     private var currentUrl: String = ""
     private val visitedUrls = mutableListOf<String>()
 
     @SuppressLint("WrongThread")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        
+
         val localBroadcastManager = LocalBroadcastManager.getInstance(this)
         localBroadcastManager.registerReceiver(
             closeReceiver,
             IntentFilter("com.opacitylabs.opacitycore.CLOSE_BROWSER")
         )
 
+        localBroadcastManager.registerReceiver(
+            cookiesForCurrentURLRequestReceiver,
+            IntentFilter("com.opacitylabs.opacitycore.GET_COOKIES_FOR_CURRENT_URL")
+        )
+
+        localBroadcastManager.registerReceiver(
+            cookiesForDomainRequestReceiver,
+            IntentFilter("com.oapcitylabs.opacitycore.GET_COOKIES_FOR_DOMAIN")
```

Comment:
```suggestion
            IntentFilter("com.opacitylabs.opacitycore.GET_COOKIES_FOR_DOMAIN")
```

Oh man, this typo killed me, I was wondering why the `get_cookies_for_domain` was returning NO cookies whatsoever, while `get_cookies_for_current_url` was working 😂